docs/API/defineFetch-share.md

```diff
@@ -0,0 +1,113 @@
+# `defineFetch.share` API
+
+The `share` property on 'defineFetch' allows other `defineFetch` actions to store data on the same key.
+
+A common use case for `share` are CRUD endpoints that need to update the same data store key through `get` and `put` http requests.
+
+### `share Props`
+
+---
+
+| Prop        | Type             | Default    | Description                                                                                                        |
+| ----------- | ---------------- | ---------- | ------------------------------------------------------------------------------------------------------------------ |
+| `namespace` | string           | required\* | Another `defineFetch` function sharing the same name space will result them sharing the same key to the data store |
+| `merge`     | func(prev, next) | required\* | `merge` adds customizing functionality before the data is applied to the data store.                               |
+
+&nbsp;
+
+Example:
+
+```ts
+// component/actions/getPerson.ts
+
+import defineFetch from '@sift/resift/defineFetch';
+
+export default defineFetch({
+  displayName: 'Get Person',
+  share: {
+    namespace: 'person',
+    // Keep data flow as is
+    merge: (previous, next) => ({ ...previous, ...next }),
+  },
+  make: (personName: string) => ({
+    key: [personName],
+    request: () => ({ http }) =>
+      return http({
+        method: 'GET',
+        route: `/person/${personId}`,
+      }),
+  }),
+});
+```
+
+```ts
+// component/actions/editPerson.ts
+
+import defineFetch from '@sift/resift/defineFetch';
+
+export default defineFetch({
+  displayName: 'Edit Person',
+  share: {
+    namespace: 'person',
+    merge: (previous, next) => {
+      // Customize and add data if needed
+      return { ...previous, ...next, lastEdited: new Date() }
+    },
+  },
+  make: (personName: string) => ({
+    key: [personName],
+    request: (personData: any) => ({ http }) =>
+      return http({
+        method: 'PUT',
+        route: `/person/${personId}`,
+        data: personData
+      }),
+  }),
+});
+```
+
+```tsx
+// Example container using `definefetch` above
```

Comment:
@ricokahler, what do you recommend?

Reply:
I think it makes sense to just call this component `<Person />` or `<Profile />`. Instead of putting this logic in a container component, let's just make the example put it in the component itself.

Unless they've used redux before, I just don't think people will know what a container component is. And from the mouth of the prophet himself:

> Update from 2019: I wrote this article a long time ago and my views have since evolved. In particular, I don’t suggest splitting your components [into container and presentational components] anymore. If you find it natural in your codebase, this pattern can be handy. But I’ve seen it enforced without any necessity and with almost dogmatic fervor far too many times. The main reason I found it useful was because it let me separate complex stateful logic from other aspects of the component. Hooks let me do the same thing without an arbitrary division. This text is left intact for historical reasons but don’t take it too seriously.

[link](https://medium.com/@dan_abramov/smart-and-dumb-components-7ca2f9a7c7d0)